apps/condo/domains/organization/components/EmployeeRolesSettingsContent.tsx

```diff
@@ -1,62 +1,629 @@
+            newPermissions = { ...oldPermissions, [permissionKey]: newValue, [readPermissionKey]: true }
+        } else {
+            newPermissions = { ...oldPermissions, [permissionKey]: newValue }
+        }
+    }
+
+    return newPermissions
+}
+
+/**
+ * Checks that certain checked checkbox in table is disabled.
+ * If there are no one of the other employee roles who have that permission, then checkbox will be disabled.
+ * If it's "canRead" checkbox, then also checks such logic for "canManage" permissions of this group.
+ */
+const isCheckboxDisabled = ({ checkboxValue, employeeRoleId, permissionKey, isReadPermission, permissionsState, pathToPermissionsGroup }) => {
+    if (!checkboxValue) return false
+
+    const otherEmployeeRoleIds = Object.keys(omit(permissionsState, employeeRoleId))
+    const pathToPermissionsGroupFromEmployeeRole = pathToPermissionsGroup.slice(1)
+    let checkboxDisabled = checkboxValue
+
+    for (const roleId of otherEmployeeRoleIds) {
+        const isOtherEmployeeHasPermission = get(permissionsState, [roleId, ...pathToPermissionsGroupFromEmployeeRole, permissionKey], false)
+
+        if (isOtherEmployeeHasPermission) {
+            checkboxDisabled = false
+            break
+        }
+    }
+
+    if (isReadPermission) {
+        const permissionsGroup = get(permissionsState, pathToPermissionsGroup)
+        checkboxDisabled = false
+
+        for (const permission of Object.keys(permissionsGroup)) {
+            let otherEmployeeHasPermission = false
+
+            for (const roleId of otherEmployeeRoleIds) {
+                const isEmployeeHasPermission = get(permissionsState, [roleId, ...pathToPermissionsGroupFromEmployeeRole, permission], false)
+
+                if (isEmployeeHasPermission) {
+                    otherEmployeeHasPermission = true
+                    break
+                }
+            }
+
+            if (!otherEmployeeHasPermission) {
+                checkboxDisabled = true
+                break
+            }
+        }
+    }
+
+    return checkboxDisabled
+}
+
+/**
+ * Returns a message notifying that the role update is complete.
+ * If there is only a checkbox withdrawal (removal of permissions), then there will be a DeletePermissionsMessage message,
+ * in another case (adding permissions/adding and deleting permissions) will be a UpdatePermissionsMessage message.
+ */
+const getSaveNotificationMessage = ({ intl, initialPermissionsState, permissionsState }): string => {
+    const UpdatePermissionsMessage = intl.formatMessage({ id: 'pages.condo.settings.employeeRoles.notification.updatePermissions' })
+    const DeletePermissionsMessage = intl.formatMessage({ id: 'pages.condo.settings.employeeRoles.notification.deletePermissions' })
+
+    let notificationMessage = DeletePermissionsMessage
+
+    for (const roleId of Object.keys(initialPermissionsState)) {
```

Comment:
Triple `for` just for notification diff checking seems impressive :)

Reply:
We check `permission` in the `group of permissions` (appId for b2bApps) in the `employee role`.
I think it's ok